lib/commands/preload.ts

```diff
@@ -305,6 +304,33 @@ Can be repeated to add multiple certificates.\
 		}
 	}
 
+	private async checkImageFile(image: string, deviceType?: string) {
+		try {
+			const fs = await import('fs');
+			await fs.promises.access(image);
+		} catch (error) {
+			throw new ExpectedError(`Cannot access image file: "${image}"`);
+		}
+		const path = await import('path');
+		const zipMsg =
+			'The image file must be unzipped unless the device type is Intel Edison';
+		const isZip = path.extname(image) === '.zip';
+		if (deviceType) {
+			if (deviceType === 'intel-edison' && !isZip) {
+				throw new ExpectedError(
+					'Error: The Intel Edison device type requires the image to be a zip file',
```

Comment:
Oh, I think this validation is broken. Using the web dashboard, I just realised that an `intel-edison` _app_ also allows downloading images for the 'qemu x86 32bit' device type, and in this case, the zip file must be unzipped even though the app type is `intel-edison`.

Reply:
I've amended my commit to cope with 'qemu x86 32bit' devices (non zip image files) in 'intel-edison' apps. Still needs your review and squashing. :-)